src/pudl/metadata/helpers.py

```diff
@@ -590,3 +591,28 @@ def groupby_aggregate(  # noqa: C901
     # Enforce original data types, which nulls and errors may have changed
     result = result.astype(dtypes, copy=False)
     return result, reports
+
+
+# --- Jinja Filters --- #
+
+
+def _format_resource_name_cross_ref(resource_name: str) -> str:
+    """Format resource name as a cross_ref for sphinx docs.
+
+    Sphinx throws an error when creating a cross ref for
+    a resource that has a preceeding underscore. It is
+    also possible for resources to have identical names
+    when the preceeding underscore is removed. This funcntion
+    adds a preceeding 'i' to cross references for resources
+    with preceeding underscores.
+
+    Args:
+        resource_name: the name of the resource.
+
+    Returns:
+        The name of the resource formatted for a sphinx cross ref.
+    """
+    return re.sub(r"^_.*", lambda x: "i" + x.group(0), resource_name)
```

Comment:
This function is only applied to the resource names when generating the .rst files for the documentation. The thing that is getting reformatted here is the sphinx target which are specified using a preceding underscore:

```rst
.. _My target:

Explicit targets
~~~~~~~~~~~~~~~~

Reference `My target`_.
```

Sphinx throws a "malformed target" error with the intermediate tables because they already have a preceding underscore. The work around here is to just add a preceding "i" to the target:

```rst
.. _i_out_eia__monthly_capacity_factor_by_generator:

-------------------------------------------------------------------------------
_out_eia__monthly_capacity_factor_by_generator
-------------------------------------------------------------------------------

Monthly estimates of generator capacity factor. Capacity factor is calculated...
```
The `i_` is not rendered anywhere in the docs, just the .rst files.

Reply:
Great, thanks for clarifying!